pkg/controller/servicebindingrequest/sbrcontroller.go

```diff
@@ -113,18 +113,32 @@ func updateFunc(logger *log.Log) func(updateEvent event.UpdateEvent) bool {
 		isConfigMap := isOfKind(e.ObjectNew, "ConfigMap")
 
 		if isSecret || isConfigMap {
-			dataFieldsAreEqual, err := compareObjectFields(e.ObjectNew, e.ObjectOld, "data")
+			dataFieldsAreEqual, err := compareObjectFields(e.ObjectOld, e.ObjectNew, "data")
 			if err != nil {
 				logger.Error(err, "error comparing object fields: %s", err.Error())
 				return false
 			}
-
 			logger.Debug("Predicate evaluated", "dataFieldsAreEqual", dataFieldsAreEqual)
 			return !dataFieldsAreEqual
 		}
+		logger.Debug("updateFunc", "Predicate evaluated for Kind", e.ObjectOld.GetObjectKind)
+		specsAreEqual, err := compareObjectFields(e.ObjectOld, e.ObjectNew, "spec")
+		if err != nil {
+			logger.Error(err, "error comparing object fields: %s", err.Error())
+		}
+		logger.Debug("Predicate evaluated", "Specs of the Object in predicate filtering changed =", specsAreEqual)
```

Comment:
I still find single log line easier to digest:

```
Service X of $GROUP/KIND updated, spec part: changed|unchanged, status part: changed|unchange, reconcile triggered: true|false
```

Reply:
Hi @pedjak can you suggest a function that can take in multiple values in a single logging message. I could not locate it for logger being used by the project

Reply:
How about this:

```
shouldReconcile := !specsAreEqual || !statusAreEqual
logger.Debug("Resource update event received", "GVK", e.ObjectNew.GetObjectKind(), "spec changed", !specsAreEqual, "status changed", !statusAreEqual, "should reconcile", shouldReconcile)

```

Reply:
Done https://github.com/redhat-developer/service-binding-operator/pull/597/commits/b904a2186b93b06ef06213f2a23139572628abdc